service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -176,34 +179,46 @@ fn generate_endpoint_modification(service: &Service) -> Option<String> {
 
 // IoT defines a lot of empty (and therefore unnecessary) request shapes
 // don't clutter method signatures with them
-fn generate_method_signature(operation: &Operation, shape: &Shape) -> String {
-    if shape.members.is_some() && !shape.members.as_ref().unwrap().is_empty() {
-        format!(
-            "fn {method_name}(&self, input: {input_type})",
-            method_name = operation.name.to_snake_case(),
-            input_type = operation.input_shape()
-        )
-    } else {
-        format!(
+fn generate_method_signature(operation: &Operation, shape: &Option<&Shape>) -> String {
+    match shape {
+        Some(s) => {
+            if s.members.is_some() && !s.members.as_ref().unwrap().is_empty() {
```

Comment:
If you want a little more typesafety, you could consider using a match statement like this:

```rust
match s.members {
    Some(ref members) if !members.is_empty() => {
        ...
    },
    _ => {
        ...
    }
}
```